server/server.go

```diff
@@ -405,6 +409,12 @@ func (s *Server) StartExitMonitor() {
 							}
 						}
 					}
+					if c != nil && s.watcherWaitContainer(containerID) {
+						logrus.Debugf("waiting for container or sandbox: %v", containerID)
+						if err := s.Runtime().PollUpdateStatusStopped(c, 10); err != nil {
```

Comment:
Oh yes good point, starting this as a go routine could be nice since there is no point in blocking here. The blocking behavior is only needed when calling into `WaitContainerStateStopped()`